dask/dataframe/io/parquet/arrow.py

```diff
@@ -265,149 +91,540 @@ def _index_in_schema(index, schema):
         return False  # No index to check
 
 
-class ArrowEngine(Engine):
-    @classmethod
-    def read_metadata(
-        cls,
-        fs,
-        paths,
-        categories=None,
-        index=None,
-        gather_statistics=None,
-        filters=None,
-        split_row_groups=True,
-        **kwargs,
-    ):
-        # Define the dataset object to use for metadata,
-        # Also, initialize `parts`.  If `parts` is populated here,
-        # then each part will correspond to a file.  Otherwise, each part will
-        # correspond to a row group (populated below)
-        parts, dataset = _determine_dataset_parts(
-            fs, paths, gather_statistics, filters, kwargs.get("dataset", {})
+def _get_dataset_object(paths, fs, filters, dataset_kwargs):
+    """ Generate a ParquetDataset object
+    """
+    if len(paths) > 1:
+        # This is a list of files
+        base, fns = _analyze_paths(paths, fs)
+        proxy_metadata = None
+        if "_metadata" in fns:
+            # We have a _metadata file. PyArrow cannot handle
+            #  "_metadata" when `paths` is a list. So, we shuld
+            # open "_metadata" separately.
+            paths.remove(fs.sep.join([base, "_metadata"]))
+            fns.remove("_metadata")
+            proxy_metadata = (
+                pq.ParquetDataset(
+                    fs.sep.join([base, "_metadata"]),
+                    filesystem=fs,
+                    filters=filters,
+                    **dataset_kwargs,
+                )
+                .pieces[0]
+                .get_metadata()
+            )
+        dataset = pq.ParquetDataset(
+            paths, filesystem=fs, filters=filters, **dataset_kwargs
+        )
+        if proxy_metadata:
+            dataset.metadata = proxy_metadata
+    elif fs.isdir(paths[0]):
+        # This is a directory.  We can let pyarrow do its thing.
+        allpaths = fs.glob(paths[0] + fs.sep + "*")
```

Comment:
Only the top-level files - the point being to check for _metadata. Is _metadata guaranteed to be the first path at `paths[0]`?

Reply:
`paths` is a list of length 1 (only contains the directory name).  So, we are assuming `_metadata` will be in the top-level of the directory, but we don't assume its the first path.  Am I misunderstanding the question?

Reply:
No, my mistake - I thought paths[0], below, was referring to this allpaths. It may make no practical difference, but why not `fs.exists(paths[0] + fs.sep + "_metadata")` ?

Reply:
The reason is that we actually use the `base`, and `fns` later to construct our own global metadata structure if there is no `_metadata`.  Techincally, we don't need to do this, because paths[0] is `base`, and each file name could be reconstructed from the full path (and `base`).  However, I did it this way to make the logic look similar for the different cases.

Reply:
OK, if a listing is required anyway. I can see the logic would get convoluted again... however, in some storage like S3, checking the existence and reading a single file can be much faster than listing a directory. Worth noting, maybe.

Reply:
> however, in some storage like S3, checking the existence and reading a single file can be much faster than listing a directory

Good point - I'll make a note in the code that we may want to avoid the directory listing